Sources/SystemExtras/FileOperations.swift

```diff
@@ -576,6 +576,21 @@ extension FileDescriptor {
     return .success(DirectoryStream(rawValue: dirp))
     #endif
   }
+
+  public func sync() throws {
+    return try _sync().get()
+  }
+
+  internal func _sync() -> Result<Void, Errno> {
+    #if os(Windows)
+    // TODO: Implement by `FlushFileBuffers`?
+    return .failure(Errno(rawValue: ERROR_NOT_SUPPORTED))
+    #else
+    nothingOrErrno(retryOnInterrupt: false) {
+      system_fsync(self.rawValue)
```

Comment:
I should use `fcntl(fd, F_FULLFSYNC)` instead of `fsync(fd)` on Darwin.

- https://github.com/golang/go/issues/26650
- https://github.com/rust-lang/rust/blob/f1bc669636023c8643602431791c7f26e5a6edef/library/std/src/sys/fs/unix.rs#L1210

Reply:
31bc63998d8c7d97d2c642b0a1dfde68e4d94f9c